src/hotspot/share/cds/archiveBuilder.cpp

```diff
@@ -232,6 +233,9 @@ void ArchiveBuilder::gather_klasses_and_symbols() {
   log_info(cds)("Gathering classes and symbols ... ");
   GatherKlassesAndSymbols doit(this);
   iterate_roots(&doit);
+  if (DynamicDumpSharedSpaces) {
+    iterate_primitive_array_klasses(&doit);
```

Comment:
This call can be moved in `DynamicArchiveBuilder::iterate_roots`. That way, you don't need the dummy implementation of `StaticArchiveBuilder::iterate_primitive_array_klasses`.

Reply:
Fixed